com.hazelcast.jet.core.ExecutionLifecycleTest.when_processorCooperativeCompleteThrows_then_failJob

https://hazelcast-l337.ci.cloudbees.com/view/Jet/job/jet-sonar/com.hazelcast.jet$hazelcast-jet-core/1192/testReport/junit/com.hazelcast.jet.core/ExecutionLifecycleTest/when_processorCooperativeCompleteThrows_then_failJob/

```
Error Message
expected:<8> but was:<4>
Stacktrace
java.lang.AssertionError: expected:<8> but was:<4>
	at com.hazelcast.jet.core.ExecutionLifecycleTest.assertPClosedWithError(ExecutionLifecycleTest.java:525)
	at com.hazelcast.jet.core.ExecutionLifecycleTest.when_processorCooperativeCompleteThrows_then_failJob(ExecutionLifecycleTest.java:293)
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/ExecutionLifecycleTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/ExecutionLifecycleTest.java
@@ -252,7 +252,7 @@ public class ExecutionLifecycleTest extends JetTestSupport {
         Job job = runJobExpectFailure(dag, e);
 
         // Then
-        assertPClosedWithError(e, true);
+        assertPClosedWithError(e);
         assertPsClosedWithError(e);
         assertPmsClosedWithError(e);
         assertJobFailed(job, e);
@@ -272,7 +272,7 @@ public class ExecutionLifecycleTest extends JetTestSupport {
         Job job = runJobExpectFailure(dag, e);
 
         // Then
-        assertPClosedWithError(e, false);
+        assertPClosedWithError(e);
         assertPsClosedWithError(e);
         assertPmsClosedWithError(e);
         assertJobFailed(job, e);
@@ -290,7 +290,7 @@ public class ExecutionLifecycleTest extends JetTestSupport {
         Job job = runJobExpectFailure(dag, e);
 
         // Then
-        assertPClosedWithError(e, false);
+        assertPClosedWithError(e);
         assertPsClosedWithError(e);
         assertPmsClosedWithError(e);
         assertJobFailed(job, e);
@@ -308,7 +308,7 @@ public class ExecutionLifecycleTest extends JetTestSupport {
         Job job = runJobExpectFailure(dag, e);
 
         // Then
-        assertPClosedWithError(e, false);
+        assertPClosedWithError(e);
         assertPsClosedWithError(e);
         assertPmsClosedWithError(e);
         assertJobFailed(job, e);
@@ -519,11 +519,7 @@ public class ExecutionLifecycleTest extends JetTestSupport {
         assertEquals(0, MockP.receivedCloseErrors.size());
     }
 
-    private void assertPClosedWithError(Throwable e, boolean initExpectedToFail) {
-        // can't assert initCount here, init() is not called on next processor after first processor fails
-        if (!initExpectedToFail) {
-            assertEquals(TOTAL_PARALLELISM, MockP.initCount.get());
-        }
+    private void assertPClosedWithError(Throwable e) {
         assertEquals(TOTAL_PARALLELISM, MockP.closeCount.get());
         assertEquals(TOTAL_PARALLELISM, MockP.receivedCloseErrors.size());
 

